faults/common.go

```diff
@@ -0,0 +1,124 @@
+// Package faults provides common headers and client-side fault injection functionality.
+package faults
```

Comment:
is this package meant to be internal? do we expect packages outside of baseplate.go v0 to use it?

Reply:
My understanding is that this implementation is separate from any other baseplate.go implementation, so I intend to duplicate a lot of this code exactly once for use in all non-v0 baseplate.go. Please let me know if that is not the recommended approach here.

Reply:
internal allows you to use this package inside bp.go v0 but forbids it from being used outside of bp.go v0.

Reply:
the point is that the API surface of this package is overly complicated that we are not really comfortable commit to not making any breaking changes in the future, we don't have a no-breaking-change commitment for internal packages.

Reply:
Ack, thanks. Yeah, the plan was to keep it internal since it's just referenced by the httpbp and thriftbp bp.go v0 packages. My background is in C++ so I'm still catching up on some of these Go-isms. Thanks for poking here -- moved!